repos/system_upgrade/common/actors/checkconsumedassets/libraries/check_consumed_assets.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+from leapp import reporting
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.models import ConsumedDataAsset, LeappRepositoryInfo
+
+
+def inhibit_if_assets_with_incorrect_version():
+    leapp_info = next(api.consume(LeappRepositoryInfo), None)
+    if not leapp_info:
+        raise StopActorExecutionError('Did not receive any LeappRepositoryInfo messages.')
+
+    version_re = '([0-9]*)\\.([0-9]*)\\.([0-9]*)'
+    for consumed_asset in api.consume(ConsumedDataAsset):
+        if not consumed_asset.target_leapp_version:
+            # Most likely an old file that predates the introduction of versioning to data assets
+            summary = 'The asset {0} is missing target leapp version in its metadata'.format(consumed_asset.filename)
+            reporting.create_report([
+                reporting.Title('{0} data is outdated'.format(consumed_asset.fulltext_name)),
+                reporting.Summary(summary),
+                reporting.ExternalLink(title=consumed_asset.docs_title, url=consumed_asset.docs_url),
+                reporting.Severity(reporting.Severity.MEDIUM),
+                reporting.Groups([reporting.Groups.INHIBITOR, reporting.Groups.REPOSITORY]),
+            ])
+            continue
+
+        version_matches_expected_form = re.match(version_re, consumed_asset.target_leapp_version)
+        if version_matches_expected_form:
+            # We need to parse in order to tell user more than just the asset version is wrong
+            asset_leapp_version = tuple(int(group) for group in version_matches_expected_form.groups())
+        else:
+            summary = '{0} has invalid value specifying leapp version for which it is meant.'
+            reporting.create_report([
+                reporting.Title('The leapp data file {0} is malformed'.format(consumed_asset.filename)),
+                reporting.Summary(summary.format(consumed_asset.fulltext_name)),
+                reporting.ExternalLink(title=consumed_asset.docs_title, url=consumed_asset.docs_url),
+                reporting.Severity(reporting.Severity.MEDIUM),
```

Comment:
Inhibitor report entry should always have a "HIGH" priority.

Reply:
Fixed.